### DISCUSSION TITLE
Error  dp_train

### DISCUSSION THREAD

**Discussion by vijendra037** [deepmodeling/deepmd-kit#33] [Q&A]
I have installed deepmd kit according to instruction but with tensorflow version 1.13 and python 3.7.3.
while running dp_train file I got error. I checked your troubleshoot and again install with   -DTF_GOOGLE_BIN=true in the process of cmake.
but again I  get this error.

Traceback (most recent call last):
  File "/home/vkumar/tensorflow_env_dir/deepmd_root/bin/dp_train", line 16, in <module>
    from deepmd.Model import NNPModel
  File "/home/vkumar/tensorflow_env_dir/deepmd_root/bin/../lib/deepmd/Model.py", line 21, in <module>
    op_module = tf.load_op_library(module_path + "libop_abi.so")
  File "/home/vkumar/anaconda3/envs/tensorflow_env/lib/python3.7/site-packages/tensorflow/python/framework/load_library.py", line 61, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename)
tensorflow.python.framework.errors_impl.NotFoundError: /home/vkumar/tensorflow_env_dir/deepmd_root/lib/deepmd/libop_abi.so: undefined symbol: _ZN10tensorflow12OpDefBuilder4AttrESs

please help me for this.

**Comment by njzjz** [comment]
Maybe your version of TensorFlow for Python is not consistent with that of TensorFlow for C++. Is your version of TensorFlow for C++ v1.13?

**Comment by dingye18** [comment]
It looks like the `libop_abi.so` did not find the right path to `libtensorflow_framework.so`. Did you add the `$tensorflow_root/lib` to your `LD_LIBRARY_PATH` environment parameters? 
If not, try this command:
`export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$tensorflow_root/lib`.
BTW, it is not suggested to use python3.7.3 for tensorflow.  There will be some unexpected errors when use python 3.7.3 for tensorflow. [Is python 3.7.x supported with Tensorflow](https://github.com/tensorflow/tensorflow/issues/17022)

**Comment by Paul-St-Young** [comment]
I encountered a similar error, but via a different route, using Python 3.6.8 and Watson Machine Learning conda environment v1.6.1.

```dp -> from deepmd.main import main
deepmd/__init__.py:2 -> from .DeepEval   import DeepEval
DeepEval.py:18 -> op_module = tf.load_op_library(os.path.join(module_path, "libop_abi.{}".format(ext)))
File "/opt/apps/anaconda3/envs/wmlce-v1.6.1-py3.6/lib/python3.6/site-packages/tensorflow/python/framework/load_library.py", line 61, in load_op_library
lib_handle = py_tf.TF_LoadLibrary(library_filename)
tensorflow.python.framework.errors_impl.NotFoundError:
.local/lib/python3.6/site-packages/deepmd/libop_abi.so: undefined symbol: _ZN10tensorflow12OpDefBuilder4AttrESs
```

**Comment by Paul-St-Young** [comment]
@dingye18  `libtensorflow_framework.so` is in my `LD_LIBRARY_PATH`.
`nm libtensorflow_framework.so | grep _ZN10tensorflow12OpDefBuilder4` shows
`0000000000680a98 t 0000007c.plt_call._ZN10tensorflow12OpDefBuilder4AttrENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE'
'0000000000e94670 T _ZN10tensorflow12OpDefBuilder4AttrENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE`

Basically, `OpDefBuilder4AttrENSt7` is in my tensorflow library, but `libop_abi.so` is looking for `OpDefBuilder4AttrESs`.
I'm trying to use `DeePMD 1.1.4` with `tensorflow 1.14.0` distributed inside `wmlce-v1.6.1`.
Is there an expected incompatibility? How can I circumvent this error?

**Comment by njzjz** [comment]
> @dingye18 `libtensorflow_framework.so` is in my `LD_LIBRARY_PATH`.
> `nm libtensorflow_framework.so | grep _ZN10tensorflow12OpDefBuilder4` shows
> `0000000000680a98 t 0000007c.plt_call._ZN10tensorflow12OpDefBuilder4AttrENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE' '0000000000e94670 T _ZN10tensorflow12OpDefBuilder4AttrENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE`
> 
> Basically, `OpDefBuilder4AttrENSt7` is in my tensorflow library, but `libop_abi.so` is looking for `OpDefBuilder4AttrESs`.
> I'm trying to use `DeePMD 1.1.4` with `tensorflow 1.14.0` distributed inside `wmlce-v1.6.1`.
> Is there an expected incompatibility? How can I circumvent this error?

Please check if you use a different gcc version for tensorflow and deepmd-kit (for example 4.8 and 5.4). Please use the same compiler version.
